src/hooks/useTimer.ts

```diff
@@ -1,10 +1,8 @@
-import { useEffect, useState } from 'react';
+import { useEffect, useMemo, useState } from 'react';
 
-// TODO: Abstract this as a props
-const mintDate = new Date('2022-05-23T00:00:00');
-
-export default function useTimer() {
+export default function useTimer(date: string) {
```

Comment:
question: is this function deterministic?

in other words, if i pass in a hard-coded `date`, will it always return the same `timestamp` and `hasEnded` output? my guess is no, because this hook calls `Date.now()`, which means it depends on when the test runs...

i wonder if this would be easier to test if we passed in both `mintDate` and `currentTime`:
```js
useTimer(mintDate, currentTime) {}
```

then we know what the initial output should be, and directly compare strings, like:
```js
expect(timestamp).toEqual("00:12:34:56")
// simulate moving forward 1 second
expect(timestamp).toEqual("00:12:34:55")
// simulate moving forward 24 hours
expect(hasEnded).toEqual(true)
```

Reply:
Make sense 🤔 Definetely much easier to test this way. 

I've refactor the `useTimer` hooks to support `currentTime` as a second params (optional). If its not available, will set to current date

Thanks for the string comparison idea! Felt much confident with it 😆